M2/Macaulay2/d/actors4.d

```diff
@@ -992,10 +985,13 @@ tostringfun(e:Expr):Expr := (
      is x:RRcell do toExpr(tostringRR(x.v))
      is x:RRicell do toExpr(tostringRRi(x.v))
      is z:CCcell do toExpr(tostringCC(z.v))
-     is Error do toExpr("<<an error message>>")
+     is err:Error do toExpr(err.message)
      is Sequence do toExpr("<<a sequence>>")
      is HashTable do toExpr("<<a hash table>>")
-     is List do toExpr("<<a list>>")
+     is x:List do (
+	 if ancestor(x.Class, filePositionClass)
+	 then toExpr(tostringFilePosition(e))
+	 else toExpr("<<a list>>"))
```

Comment:
Not to add to your work but would it be simpler if FilePosition was its own interpreter-level class?